src/Update.cpp

```diff
@@ -1246,6 +1251,23 @@ void DoPlaceClose(int i, int j, unsigned short int ts, int tn, int DoAnyway)
 	}
 }
 
+void UpdateHostClosure() {
+    // Single-threaded here, because the same host index might come up in
+	// different threads. (Which is the point of this buffered update...)
```

Comment:
This could be done multi-threaded, like `inf_queue`, but it's probably not worth the complexity unless profiling suggests it's a hotspot.

Reply:
I don't think it'll be a hotspot - not enough work with just those two lines... 

I wondered about having all the threads traversing through the whole host_closure_queue together, but each of them only updating if (host_index % num_threads) == thread_no ...

Reply:
Tested with parallelise and is was a bit slower! So reverted...